lib/presentation/blocs/abstract_location_cubit.dart

```diff
@@ -0,0 +1,24 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:movna/domain/entities/gps_coordinates.dart';
+import 'package:movna/domain/entities/location.dart';
+import 'package:movna/domain/faults.dart';
+
+/// Abstract class representing a cubit that can keep track of the device
+/// location.
+///
+/// Implementing cubits' states must implement [AbstractLocationState].
+abstract class AbstractLocationCubit<T extends AbstractLocationState>
+    extends Cubit<T> {
+  AbstractLocationCubit(super.initialState);
+}
+
+abstract interface class AbstractLocationState {
+  GpsCoordinates? get coordinates;
+  Location? get location;
+  LocationStateType get type;
+  Fault? get fault;
+}
```

Comment:
I don't understand this, why not having a single getter `location` or maybe a `Result<Location, Fault>` ?

`coordinates` and `location` seem redundant to me, and I don't understand the presence of the two others